cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js

```diff
@@ -206,7 +219,10 @@ export default class AddToHydratorModal extends Component {
     // This is a hack.. should not do this
     // We are still shipping kafka-plugins with hydrator-plugins 1.7 but
     // it doesn't contain the streamingsource or batchsource plugins
-    let pluginArtifact = artifactsList.filter((artifact) => artifact.name === 'kafka-plugins');
+    let pluginArtifact = find(artifactsList, {name: 'kafka-plugins'});
+    if (!pluginArtifact) {
+      return T.translate(`${PREFIX}.kafkaPipeline.pipelineError`);
+    }
     pluginArtifact = pluginArtifact[pluginArtifact.length - 1];
```

Comment:
If you're using `find` now then you don't need this line anymore right, since it already only returns the one element?